src/components/Modal/Modal.tsx

```diff
@@ -75,7 +75,15 @@ export const Modal = ({
   };
 
   const handleEscKey = (e: KeyboardEvent<HTMLDivElement>) => {
-    e.nativeEvent.stopImmediatePropagation();
+    if (e.nativeEvent?.stopImmediatePropagation) {
+      e.nativeEvent.stopImmediatePropagation();
+    } else if ((e as unknown as Event).stopImmediatePropagation) {
+      (e as unknown as Event).stopImmediatePropagation();
+    } else if (e.stopPropagation) {
+      e.stopPropagation();
+    } else {
+      console.log("Couldn't stop Esc key press propagation.");
+    }
```

Comment:
With the type updated above then you could check for the method like this:

```suggestion
    if ("nativeEvent" in e && e.nativeEvent?.stopImmediatePropagation) {
      e.nativeEvent.stopImmediatePropagation();
    } else if ("stopImmediatePropagation" in e) {
      e.stopImmediatePropagation();
    } else if (e.stopPropagation) {
      e.stopPropagation();
    }
```

This means you don't have to cast to `unknown`. I also think we probably don't need the console.log.

Reply:
I also think this approach is a lot better and removes the weird type conversion to `unknown`. Made this change and added a test for stopImmediatePropagation as suggested in 28d767dbf25ecbf215350ac3f504b018da9ec2ba.